api/desecapi/serializers.py

```diff
@@ -143,6 +153,69 @@ def update(self, instance: RRset, validated_data):
         return instance
 
 
+class RRsetListSerializer(ListSerializer):
+    def create(self, validated_data):
+        index = {(data.get('subname', ''), data['type']) for data in validated_data}
+        for (subname, type_) in index:
+            if not index[(subname, type_)].get('type', None):
```

Comment:
What does this do? It looks like you're calling `get()` on the value of a `dict` entry, but `index` is a set, not a dict. Did you miss adding the value part in the comprehension expression that produces `index`?

Reply:
Previous comment resolved.

New thought: How can `get('type', None)` evaluate to `False`? That's only the case if `type` does not exist. However, two lines earlier, you access `data['type']`, and if `type` does not exist, this would already have raised an exception.